modules/core/shared/src/main/scala/eu/timepit/refined/api/Inference.scala

```diff
@@ -4,30 +4,57 @@ package eu.timepit.refined.api
  * Evidence that states if the conclusion `C` can be inferred from the
  * premise `P` or not.
  *
- * This type class is used to implement refinement subtyping. If a valid
- * `Inference[P, C]` exists, the type `F[T, P]` is considered a subtype
+ * This trait is used to implement refinement subtyping.
+ */
+trait Inference[P, C] {
+  type T[P2, C2] <: Inference[P2, C2]
```

Comment:
What is the purpose of `T`? I see that it is used in `adapt` but can't we just use `Inference` there? Subtypes should be able to override `adapt` with a more concrete type (eg. `override def adapt[P2, C2](adaptedShow: String): InferAlways[P2, C2] = ...`).

Reply:
I kept it there in case we wanted to use Dependent types for something like; `def something[C,P](f: Inference[C,P]): f.T[C,P] = f...`
However in the end I didn't find a way to use it for the implicit conversion. So if that's unnecessary any longer I can do it as suggested